synchronizations/Synchronization.md

```diff
@@ -0,0 +1,223 @@
+
+
+Then, our Program.cs looks like this.
+```csharp
+class Program
+    {
+        private FamilyRegister familyRegister;
+        private PersonRegister personRegister;
+
+        private ModelRepository repository = new ModelRepository();
+
+        private Families2PersonsSynchronization synchronization = new Families2PersonsSynchronization();
+
+        public Program()
+        {
+            var familyRootModel = new Model();
+            var personRootModel = new Model();
+
+            familyRegister = new FamilyRegister();
+            personRegister = new PersonRegister();
+
+            synchronization.Synchronize(
+                synchronization.SynchronizationRule<Families2PersonsSynchronization.FamilyRegisterToPersonRegister>(),
+                ref familyRegister, // Familyregister is the left model
+                ref personRegister, // Personregister is the right model
+                SynchronizationDirection.LeftWins, // prioritizes the left synchronization of left model (without shouldCorrespond it has no impact)
+                ChangePropagationMode.TwoWay); // TwoWay means Bidirectional synchronization 
+
+            familyRootModel.RootElements.Add(familyRegister);
+            personRootModel.RootElements.Add(personRegister);
+
+            repository.Models.Add(new Uri("ttc:source"), familyRootModel);
+            repository.Models.Add(new Uri("ttc:target"), personRootModel);
+        }
+
+        static void Main(string[] args)
+        {}
+    }
+```
+
+## Synchronization
+
+As described above, in this section, we specifically address the synchronization rule/block.
+
+```csharp 
+synchronization.SynchronizationRule<Families2PersonsSynchronization.FamilyRegisterToPersonRegister>()
+```
+
+The line signifies that a synchronization rule is being invoked, with its type being `Families2PersonsSynchronization.FamilyRegisterToPersonRegister`. This rule leverages the synchronization block, which is defined within the Families2PersonsSynchronization class.
+
+Lets take a closer a look at the synchronization block. 
+```csharp
+public class FamilyRegisterToPersonRegister : SynchronizationRule<FamilyRegister, PersonRegister>
+        {
+            // Synchronization Block
+            public override void DeclareSynchronization()
+            {
+                SynchronizeMany(SyncRule<MemberToMember>(),
+                    fam => new FamilyMemberCollection(fam),
+                    persons => persons.Persons);
+            }
+        }
+```
+This method named [SynchronizeMany](../synchronizations/api/NMF.Synchronizations.SynchronizationRuleBase.yml) facilitates bidirectional synchronization, implying that the method itself defines a dependency between both models. It synchronizes dependent elements using a synchronization rule, accepting a rule, left and right selectors in the model, and the types of dependent elements. Then, it initiates synchronization using the provided parameters.
```

Comment:
It is actually a synchronization block, consisting of two lenses and a reference to a synchronization rule